src/Makefile.am

```diff
@@ -289,6 +294,32 @@ if USE_ASM
 crypto_libbitcoin_crypto_a_SOURCES += crypto/sha256_sse4.cpp
 endif
 
+crypto_libbitcoin_crypto_sse41_a_CXXFLAGS = $(CXXFLAGS)
+crypto_libbitcoin_crypto_sse41_a_CPPFLAGS = $(CPPFLAGS)
+if ENABLE_SSE41
+crypto_libbitcoin_crypto_sse41_a_CXXFLAGS += $(SSE41_CXXFLAGS)
+crypto_libbitcoin_crypto_sse41_a_CPPFLAGS += -DENABLE_SSE41
+endif
+crypto_libbitcoin_crypto_sse41_a_SOURCES = crypto/sha256_sse41.cpp
+
+crypto_libbitcoin_crypto_avx2_a_CXXFLAGS = $(CXXFLAGS)
+crypto_libbitcoin_crypto_avx2_a_CPPFLAGS = $(CPPFLAGS)
+if ENABLE_AVX2
+crypto_libbitcoin_crypto_avx2_a_CXXFLAGS += $(AVX2_CXXFLAGS)
+crypto_libbitcoin_crypto_avx2_a_CPPFLAGS += -DENABLE_AVX2
```

Comment:
Can't compiler defines be used instead so that non-autoconf builds can compile with optims too?

Specifically for this one, rather than ```ifdef(ENABLE_AVX2)```, use ```ifdef(__AVX2__)```